### PR TITLE
82 m break down movements into discrete intervals and pause resume the queue test for precision

### PR DESCRIPTION
**Pull request by andySigler**
This PR:

1. breaks down `motor.move_head()` commands into smaller steps so they can be paused
2. enqueues `motor.move_head()` commands, which can be paused, resumed, and stopped
3. refactored `robot.pause()`, `robot.resume()`, and `robot.clear()` handle the `motor`'s queue
4. removed "relative" movements from robot and instrument

### CODE REVIEW

File: tests/opentrons_sdk/drivers/test_motor.py [hunk 1]
```diff
@@ -20,10 +21,27 @@ def setUp(self):
...
+        def _move_head():
+            self.motor.move_head(x=100)
+            done.set()
+
+        thread = Thread(target=_move_head)
+        thread.start()
+
+        self.motor.resume()
+        done.wait()
+
```
Comment (levthedev) [inline tests/opentrons_sdk/drivers/test_motor.py]: Is this supposed to not have any assertions?

Comment (andySigler) [inline tests/opentrons_sdk/drivers/test_motor.py]: nope, just added one thanks

File: opentrons_sdk/drivers/motor.py [hunk 2]
```diff
@@ -268,42 +295,68 @@ def move_plunger(self, mode='absolute', **kwargs):
...
+
+        return self.queue_move_commands(args_list, increment)
+
+    def queue_move_commands(self, args_list, step):
+        tolerance = step * 0.5
+        self.current_commands = list(args_list)
+        while self.can_move.wait():
+            if self.stopped.is_set():
+                self.resume()
+                return (False, 'Stopped')
```
Comment (SimplyAhmazing) [inline opentrons_sdk/drivers/motor.py]: This functions name doesn't really seem to reflect it does.. Like maybe it should be called `consume_move_commands` or `send_move_commands` or `run` or some kind of verb. Or maybe `enqueue_move_commands`...

Also the return value of this message `(bool, error_code)` I think we should note these error codes somewhere, maybe class values, and return them via like `(bool, self.SUCCESS_CODE)`

File: opentrons_sdk/drivers/motor.py [hunk 3]
```diff
@@ -268,42 +295,68 @@ def move_plunger(self, mode='absolute', **kwargs):
...
-            for axis in 'xyz' if axis in kwargs}
+            self.wait_for_arrival(tolerance)
 
-        log.debug("MotorDriver", "Moving head: {}".format(args))
-        res = self.send_command(self.MOVE, **args)
-        return res == b'ok'
+            log.debug("MotorDriver", "Moving head: {}".format(args))
+            res = self.send_command(self.MOVE, **args)
+            if res != b'ok':
+                return (False, 'Smoothie error')
```
Comment (SimplyAhmazing) [inline opentrons_sdk/drivers/motor.py]: Related to above commend, maybe `(bool, self.SMOOTHIE_ERROR)` -- just to have a standardized place of potential error codes..

**Pull request merged**